### ISSUE TITLE
add ability to reset Brave Rewards to pre-opt in state

### ISSUE THREAD

**Issue by LaurenWags** [brave/brave-browser#1192]
## Description 
In browser-laptop we have the ability to delete a wallet and reset Brave Payments to a pre-enabled state (wallet info is removed, publisher info is removed, Brave Payments is turned off). Brave Rewards might benefit from a similar feature.

## Steps to Reproduce

(pre-req: have wallet backed up if it contains BAT)
   1. In browser-laptop, with Payments enabled, navigate to the Payments page.
   2. Click on the gear icon.
   3. Click Delete Wallet button.
   4. Click on OK.
   5. Brave Payments is disabled, wallet info is removed from profile, publisher info is removed from profile.
   6. If you re-enable at this point, you get a new wallet.

## Actual result:
No way to do this in b-c.

## Expected result:
Should be a way to get back to pre opt-in state

### Additional Information
Related b-l issues:
https://github.com/brave/browser-laptop/pull/14116 (contains additional items, not just Delete Wallet)
https://github.com/brave/browser-laptop/issues/14758

**Comment by kjozwiak** [comment]
We're going to need this implemented before the android launch IMO. Reasoning:

Because you won't be able to remove/view any files that are stored on the sandbox storage without rooting, there's really no way of resetting/getting rewards into a working state if a user ends up in a broken state. Their only choice will be to remove the entire app and re-install which isn't the best approach in terms of retention. If you want to access the sandbox storage, you'll need to root the device but this means that reward grants won't work. So it's not really feasible.

We're also getting more reports from users that have run into a state where they can't create wallets due to either having a corrupted `ledger_state` file or something else that might have happened. There's really no way of correctly recovering unless you start removing files (which Brave does not recommend doing) or uninstalling/installing. Again, not great for retention.

This feature should be disconnected from Rewards and placed somewhere in `Settings`. This way the entire Rewards feature stops working, it's still possible to nuke/restart fresh via `Settings`.

CCing @rebron @NejcZdovc @bbondy @bsclifton @LaurenWags @jenn-rhim (design).

**Comment by rebron** [comment]
cc: @anthonypkeane to put on your radar

**Comment by mandar-brave** [comment]
Add the feature to reset BR in the "Reset section"

![image](https://user-images.githubusercontent.com/35932307/53765940-4410c400-3e86-11e9-96c3-4c10beb7d8e3.png)

Text: "Reset Brave Rewards"
What it does: delete wallet and every other aspect of BR data

@jenn-rhim will create another issue to create BR Settings and separate BR panel in to a separate piece of code that is always accessible.

cc @NejcZdovc @davidtemkin

**Comment by NejcZdovc** [comment]
blocked on https://github.com/brave/brave-browser/issues/926

**Comment by rebron** [comment]
![Screen Shot 2019-03-13 at 2 29 43 PM](https://user-images.githubusercontent.com/5951041/54315835-76b07000-459c-11e9-88b6-ce639fea2bb1.png)
Should include a warning/confirm dialog similar to reset settings.

**Comment by NejcZdovc** [comment]
@rebron @mandar-brave can you please provide text for it

**Comment by jasonrsadler** [comment]
cc @mandar-brave

**Comment by mandar-brave** [comment]
Here is the dialog @jasonrsadler 

Reset Brave Rewards
This will reset all of your Brave Rewards data, including your wallet, earned advertising tokens, contribution history, wallet transaction records, recurring tips to content creators, pending contributions to content creators, and any other Brave Rewards settings you may have saved.

**Comment by mandar-brave** [comment]
On error

Resetting Brave Rewards is failing at the moment. Please restart Brave and try again shortly.
Buttons (Ok, Restart)

@jasonrsadler

**Comment by jasonrsadler** [comment]
@mandar-brave how is this:

Reset Brave Rewards
This will delete your Brave wallet and reset all of your Brave Rewards data, including your earned advertising tokens, contribution history, wallet transaction records, recurring tips to content creators, pending contributions to content creators, and any other Brave Rewards settings you may have saved.

I think we need to specify that wallet will be deleted

**Comment by rebron** [comment]
@mandar-brave @bradleyrichter 
Wondering if bullet pointing this list is better. Hard to digest in paragraph form.

Reset Brave Rewards
This will delete your Brave wallet and reset all of your Brave Rewards data, including:
  * your earned advertising tokens
  * contribution history
  * recurring tips to content creators
  * pending contributions to content creators
  * wallet transaction records
and any other Brave Rewards settings you may have saved.

**Comment by srirambv** [comment]
Probably should add a note to backup wallet for future use or reuse?

**Comment by jasonrsadler** [comment]
+1 for backup wallet, however, there are certain things like unclaimed advertising tokens that would not be recovered. cc @tmancey

**Comment by mandar-brave** [comment]
@rebron open to bullet points; not sure if the visual will look ok. @jasonrsadler i like Raf's proposal, just not sure if the dialog can take than much height and then take the buttons + error. Please comment.

**Comment by jasonrsadler** [comment]
@mandar-brave @rebron For the browser 'reset settings'. There is a link for 'Learn More'. Would 'Reset Rewards' detailed explanation be better suited for a help page as well rather than trying to squeeze all in the dialog?

**Comment by mandar-brave** [comment]
@jasonrsadler i can definitely do a FAQ question; knowing the fact that this is nuclear, we should definitely have an upfront "you are warned" thing. Well - tell us what we can squeeze and we can go from there!

**Comment by rebron** [comment]
cc: @Brave-Matt Will need updates to support docs for this one too.

**Comment by NejcZdovc** [comment]
closing as this issue is more up to date with what we want https://github.com/brave/brave-browser/issues/10064
